### DISCUSSION TITLE
breaking version upgrades?

### DISCUSSION THREAD

**Discussion by fritz-net** [berops/claudie#1438] [General] [answered]
in version 8.0 (and for some in version 8.1) the cluster had to be recreated. How high will the chance be to have those breaks in the future again? I really like the project but having a forced downtime for restoring the cluster after recreation sounds a little bit odd :(

**Comment by bernardhalas** [comment, marked as answer]
Hello, absolutely.

The cluster re-creation issue came from the fact that we were not able to use different Terraform templates within a single Claudie release. As a result, if we wanted to change the templates in a new Claudie release, the clusters had to be re-created. This was a bad design.

We're designing a new Claudie feature to allow us to use different Terraform templates within a single Claudie release. This way, we can deliver Terraform template updates independently of the Claudie release, in a rolling update manner - nodepool by nodepool.

**Reply by fritz-net** [reply]
Thanks for the super quick answer, that sounds great. So if I have 2 nodepools for a cluster it can stay online while replacing the infra beneath
